jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -290,7 +290,8 @@ public String toString() {
 
     // Use tag if not the default tag.
     if (!DEFAULT_TAG.equals(tag)) {
-      referenceString.append(':').append(tag);
+      // Append with "@tag" instead of ":tag" if tag is a digest
+      referenceString.append(tag.matches(DescriptorDigest.DIGEST_REGEX) ? '@' : ':').append(tag);
```

Comment:
I think adding a `TagType getTagType` (alternatively, `boolean isTagDigest`) would be useful for users of `ImageReference`